packages/bpk-docs/src/pages/InfiniteScrollPage/InfiniteScrollPage.js

```diff
@@ -0,0 +1,198 @@
+
+import DocsPageBuilder from './../../components/DocsPageBuilder';
+import IntroBlurb from './../../components/neo/IntroBlurb';
+
+import STYLES from './InfiniteScrollPage.scss';
+import DocsPageWrapper from '../../components/neo/DocsPageWrapper';
+
+const getClassName = cssModules(STYLES);
+
+const elementsArray = [
+  'Item 1',
+  'Item 2',
+  'Item 3',
+  'Item 4',
+  'Item 5',
+  'Item 6',
+  'Item 7',
+  'Item 8',
+  'Item 9',
+  'Item 10',
+  'Item 11',
+  'Item 12',
+  'Item 13',
+  'Item 14',
+  'Item 15',
+  'Item 16',
+  'Item 17',
+  'Item 18',
+  'Item 19',
+  'Item 20',
+];
+
+const List = ({ elements }) => (
+  <div className={getClassName('bpk-docs-infinite-scroll-page__list')}>
+    {elements.map(element => <div key={element}>{element}</div>)}
+  </div>
+);
+
+List.propTypes = {
+  elements: PropTypes.arrayOf(PropTypes.any).isRequired,
+};
+
+const FixedHeightPanel = ({ children }) => (
+  <div className={getClassName('bpk-docs-infinite-scroll-page__fixed-panel')}>
+    {children}
+  </div>
+);
+
+FixedHeightPanel.propTypes = {
+  children: PropTypes.node.isRequired,
+};
+
+const CustomSeeMore = ({ onSeeMoreClick }) => (
+  <div
+    className={getClassName('bpk-docs-infinite-scroll-page__custom-component')}
+  >
+    <BpkButton onClick={onSeeMoreClick}>See More</BpkButton>
+  </div>
+);
+
+CustomSeeMore.propTypes = {
+  onSeeMoreClick: PropTypes.func.isRequired,
+};
+
+const CustomLoading = () => (
+  <div
+    className={getClassName('bpk-docs-infinite-scroll-page__custom-component')}
+  >
+    <BpkSpinner type={SPINNER_TYPES.primary} />
+  </div>
+);
+
+const InfiniteList = withInfiniteScroll(List);
+const defaultFetch = onItemsFetch(elementsArray);
+
+const components = [
+  {
+    id: 'infinite',
+    title: 'Infinite list',
+    blurb: [],
+    examples: [
+      <FixedHeightPanel>
+        <InfiniteList
+          onItemsFetch={(index, nElements) =>
+            new Promise(resolve => {
+              setTimeout(() => {
+                const elements = [];
+                for (let i = 0; i < nElements; i += 1) {
+                  elements.push(`Item ${index + i}`);
+                }
+                resolve(elements);
+              }, 10);
+            })
+          }
+        />
+      </FixedHeightPanel>,
+    ],
+  },
+  {
+    id: 'infinite1',
+    title: 'One element per scroll',
+    blurb: [],
+    examples: [
+      <FixedHeightPanel>
+        <InfiniteList onItemsFetch={defaultFetch} elementsPerScroll={1} />
+      </FixedHeightPanel>,
+    ],
+  },
+  {
+    id: 'infiniteStop',
+    title: 'Partial load',
+    blurb: [],
+    examples: [
+      <FixedHeightPanel>
+        <InfiniteList
+          onItemsFetch={defaultFetch}
+          elementsPerScroll={5}
+          seeMoreAfter={2}
+          renderSeeMoreComponent={CustomSeeMore}
+        />
+      </FixedHeightPanel>,
+    ],
+  },
+  {
+    id: 'infiniteCustomLoading',
+    title: 'Using a custom loadingComponent',
```

Comment:
_Using a custom loading component_ would be better title, we want to keep it in plain English.